message_tagging_service/tagging_service.py

```diff
@@ -408,11 +411,27 @@ def handle(rule_defs, event_msg):
             logger.info('Tag build %s with tag(s) %s', nvr, ', '.join(dest_tags))
             tag_tasks_info = tag_build(nvr, dest_tags, koji_session)
 
-            if not tag_tasks_info:
+            failed_tasks = [item for item in tag_tasks_info if item[1] is None]
+
+            if len(failed_tasks) == len(dest_tags):
+                logger.warning(
+                    'None of tag(s) %r is applied to build %s successfully.',
+                    dest_tags, nvr)
+            else:
                 logger.warning(
-                    'None of tag(s) %r is tagged to build %s. Skip to send message.',
-                    [tag_name for tag_name, _ in tag_tasks_info], nvr)
-                continue
+                    'Tag(s) %r should be applied to build %s. But some of them,'
```

Comment:
If I read the code well, in case there is 0 failed_tasks, you still print this warning. I think you should change the `else` to `elif len(failed_tasks) != 0:`

Reply:
Done. Thanks for catching this problem. :)